azavea/raster-vision
# Add code formatter, style tests, and PR template

## Overview

This PR adds automatic code formatting using https://github.com/google/yapf. Running `scripts/format_code` will modify Python files inline to comply with pep8 style. This also adds `scripts/style_tests` which is run by CI, and a PR template.

### Checklist

- [x] Ran scripts/format_code and commit any changes

### Notes

I also looked at https://github.com/ambv/black and https://github.com/hhatto/autopep8, and went with yapf because it seemed the most popular and is configurable. I don't have strong opinions about how the styling should happen, so I didn't spend too much time researching these. 

There are some things `yapf` can't do automatically which have to be fixed manually such as unused imports and reflowing PyDocs to be < 80 chars wide.

## Testing Instructions

* Rebuild Docker image to get new dependency.
* Add some extraneous new lines to a Python source file.
* Run `./scripts/style_tests` and see that it fails.
* Run `./scripts/format_code` and see changes.

Closes #317
Closes #300
